### PR TITLE
Konnector Orange mesinfos

### PR DESCRIPTION
**Pull request by jacquarg**
Part of the MesInfos projet, it allows Orange (France) clients to fetch: 
* Geolocation ofthe cell phone
* Phone Call Log
* VoD Log.

### CODE REVIEW

**Comment by jacquarg** [comment]
Hi @ZeHiro , @nicofrand , @Phyks !
Did you have some time to look at this PR : ) ?

Thank you and Merry Xmas ; )

File: client/app/locales/en.coffee [hunk 1]
```diff
@@ -9,6 +9,8 @@ module.exports =
     'file error': 'The file could not be created/modified/deleted'
     'internal error': 'An error occured in your Cozy'
     'no bills retrieved': 'No bills retrieved'
+    'not mobile token': 'To get these data, you have to sing-in with your mobile phone number !'
```
Comment (ZeHiro) [inline client/app/locales/en.coffee]: Can you finish the line with a normal dot instead of !

File: client/app/locales/fr.coffee [hunk 2]
```diff
@@ -9,6 +9,8 @@ module.exports =
     'file error': "Le fichier n'a pas pu être créé/modifié/supprimé"
     'internal error': "Une erreur est survenue dans votre Cozy"
     'no bills retrieved': 'Pas de factures trouvées'
+    'not mobile token': 'Pour obtenir ces données, vous devez vous identifier avec votre numéro de téléphone mobile !'
```
Comment (ZeHiro) [inline client/app/locales/fr.coffee]: No "!" but a regular "." please at the end of the sentence

File: server/controllers/konnectors.coffee [hunk 3]
```diff
@@ -116,4 +118,22 @@ module.exports =
...
+            window.opener.location.reload();
+            setTimeout(function() {
+                window.close();
+            }, 500);
+        };
+    </script>
+</body>
+</html>
+"""
+#            res.redirect '../../../#konnector/' + req.konnector.slug
```
Comment (ZeHiro) [inline server/controllers/konnectors.coffee]: Can you remove the comment?

File: server/controllers/konnectors.coffee [hunk 4]
```diff
@@ -116,4 +118,22 @@ module.exports =
         req.konnector.updateFieldValues { accounts: accounts }, (err) ->
             return next err if err
 
-            res.redirect '../../../#konnector/' + req.konnector.slug
+            res.status(200).send """<!DOCTYPE html>
```
Comment (ZeHiro) [inline server/controllers/konnectors.coffee]: Will this be sent all the time ?

File: server/konnectors/orange_mobile.js [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+    connector.logger.error(`Unexpected token format: ${e}`);
+    next('token not found');
+  }
+}
+
+function requestOrange(uri, token, callback) {
+  connector.logger.info(uri);
+
+  request.get(uri, { auth: { bearer: token }, json: true }, (err, res, body) => {
+    if (res.statusCode !== 200 && res.statusCode !== '200') {
```
Comment (ZeHiro) [inline server/konnectors/orange_mobile.js]: You could use Math.parseInt(res.statusCode)

File: server/konnectors/orange_mobile.js [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+          chipType: cra.typ_units,
+          longitude: cra.loc ? cra.loc[0] : undefined,
+          latitude: cra.loc ? cra.loc[1] : undefined,
+          networkType: cra.net_lbl,
+          type: cra.desc,
+          endCause: cra.end_cause,
+        });
+      } catch (e) {
+        connector.logger.error('While parsing CRA.');
+        connector.logger.error(e);
```
Comment (ZeHiro) [inline server/konnectors/orange_mobile.js]: can you `next(e)` ?

File: server/konnectors/orange_mobile.js [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+function saveFieldsInKonnector(requiredFields, entries, data, next) {
+  connector.logger.info('saveFieldsInKonnector');
+
+  // Disable eslint because we can't require models/konnector at the top
+  // of this file (or Konnector will be empty). It's because in the require
+  // tree of models/konnector, there is the current file.
+  //eslint-disable-next-line
+  const Konnector = require('../models/konnector');
+
+  Konnector.all((err, konnectors) => {
```
Comment (ZeHiro) [inline server/konnectors/orange_mobile.js]: You can use the Konnector.bySlug request for that.

File: server/konnectors/orange_vod.js [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+    next('token not found');
+  }
+}
+
+
+function requestOrange(uri, token, callback) {
+  connector.logger.info(uri);
+
+  request.get(uri, { auth: { bearer: token }, json: true }, (err, res, body) => {
+    if (res.statusCode !== 200 && res.statusCode !== '200') {
```
Comment (ZeHiro) [inline server/konnectors/orange_vod.js]: same as for other konnector

File: server/konnectors/orange_vod.js [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+function saveFieldsInKonnector(requiredFields, entries, data, next) {
+  connector.logger.info('saveFieldsInKonnector');
+
+  // Disable eslint because we can't require models/konnector at the top
+  // of this file (or Konnector will be empty). It's because in the require
+  // tree of models/konnector, there is the current file.
+  //eslint-disable-next-line
+  const Konnector = require('../models/konnector');
+
+  Konnector.all((err, konnectors) => {
```
Comment (ZeHiro) [inline server/konnectors/orange_vod.js]: You can use Konnector.bySlug

File: server/controllers/konnectors.coffee (hunk 4 above)
Comment (jacquarg) [inline server/controllers/konnectors.coffee]: Only during the configuration of the konnector : it's the response from the redirect route ; the script just close the window which allows to identify and authenticate.

File: server/konnectors/orange_mobile.js (hunk 6 above)
Comment (jacquarg) [inline server/konnectors/orange_mobile.js]: I prefer not to. The aim is to have a continue on error, against some format unexpections.

**Comment by ZeHiro** [comment]
OK. Nice. Thanks @jacquarg

**Comment by jacquarg** [comment]
GREAT ! : ) Thanks!

File: server/konnectors/orange_mobile.js (hunk 6 above)
Comment (ZeHiro) [inline server/konnectors/orange_mobile.js]: OK

File: server/controllers/konnectors.coffee (hunk 4 above)
Comment (ZeHiro) [inline server/controllers/konnectors.coffee]: OK

**Pull request merged**